### PR TITLE
Allow /30 machine cidr in SNO

### PR DESCRIPTION
**Pull request by tsorya**
# Assisted Pull Request

## Description
The idea of sno is to minimize resources required by customers to run openshift.  We should allow the minimal cidr size as possible. 

## List all the issues related to this PR

- [ ] New Feature
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Assignees

/cc @
/cc @

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] Reviewers have been listed
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- [ ] Are the title and description (in both PR and commit) meaningful and clear?
- [ ] Is there a bug required (and linked) for this change?
- [ ] Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: internal/network/cidr_validations.go [hunk 1]
```diff
@@ -58,8 +61,12 @@ func VerifyClusterOrServiceCIDR(cidrStr string) error {
 	return verifySubnetCIDR(cidrStr, MinMaskDelta)
 }
 
-func VerifyMachineCIDR(cidrStr string) error {
-	return verifySubnetCIDR(cidrStr, MinMachineMaskDelta)
+func VerifyMachineCIDR(cidrStr string, sno bool) error {
```
Comment (romfreiman) [inline internal/network/cidr_validations.go:64]: isSNO

**Review by flaper87 — CHANGES_REQUESTED** [review]
In addition to Rom's comment (rename param to `isSNO` instead of using `sno`) I think it would be useful to expand on the PR description on why this is needed.

The current description says we *should* do this but it doesn't elaborate on why, which is important for the logs, history, and helping others learn from the work being done :)

File: internal/bminventory/inventory.go [hunk 2]
```diff
@@ -2883,7 +2883,7 @@ func (b *bareMetalInventory) updateNetworks(db *gorm.DB, params installer.V2Upda
 
 	if params.ClusterUpdateParams.MachineNetworks != nil {
 		for _, machineNetwork := range params.ClusterUpdateParams.MachineNetworks {
-			if err = network.VerifyMachineCIDR(string(machineNetwork.Cidr)); err != nil {
+			if err = network.VerifyMachineCIDR(string(machineNetwork.Cidr), common.IsSingleNodeCluster(cluster)); err != nil {
```
Comment (eranco74) [inline internal/bminventory/inventory.go:2886]: What's this syntax?
`,` between conditions

**Comment by tsorya** [comment]
/retitle Bug 2045860: Allow /30 machine cidr in SNO

**Comment by tsorya** [comment]
?

**Pull request merged**